### ISSUE TITLE
About customizing learning data

### ISSUE THREAD

**Issue by chanrai** [tech-srl/code2seq#70]
I want to modify JavaExtractor and export the label information with the project name and file name instead of the label about the method. For example, gradle|Zip.java. 

By machine learning, I'd like to get the information about code snippet.

Can you give me some advice on which files in the JavaExtractor code should be modified to output such training data?

such as 
[label](https://github.com/tech-srl/code2vec/blob/master/JavaExtractor/JPredict/src/main/java/JavaExtractor/FeaturesEntities/ProgramFeatures.java#L21)

Thanks and Best regards.

**Comment by urialon** [comment]
Hi @chanrai ,
Thank you for your interest in code2seq!

If I understand correctly, you want to predict the project name instead of
the method name?

**Comment by chanrai** [comment]
Thank you for your reply.

>If I understand correctly, you want to predict the project name instead of
the method name?

Yes, that's right.

The purpose is to get the information about OSS from the source code snippet.

Thanks and Best regards.

**Comment by chanrai** [comment]
Thank you for the polite explanation.

The Information such as file names can now be registered to labels without problems. Than you very much!

I'm an AI newbie, so I'm not sure, but when I create a model from a dataset which created from a filename, it is much smaller than the number of models created by the method.

In the first place, I don't really understand the differences between model_iter1.dict and model_iter2.dict and so on. 
Could you tell me about that?

Thanks and Best regards.

**Comment by urialon** [comment]
Hi @chanrai ,
This is not an error - it is success actually :-) 

Training stops automatically after 10 iterations over the training data in which there was no improvement over the validation set. See also: [https://machinelearningmastery.com/early-stopping-to-avoid-overtraining-neural-network-models/](https://machinelearningmastery.com/early-stopping-to-avoid-overtraining-neural-network-models/)

So in your case, the first "epoch" (iteration) performed best.
It gets 96% [precision](https://en.wikipedia.org/wiki/Precision_and_recall) which is very good, but might hint that the problem is too easy.

**Comment by chanrai** [comment]
Thank you for your reply.

>Training stops automatically after 10 iterations over the training data in which there was no improvement over the validation set.

I see! I understand.

>See also: https://machinelearningmastery.com/early-stopping-to-avoid-overtraining-neural-network-models/

Thank you for providing information. It was very helpful.

>but might hint that the problem is too easy.

Thank you for your comment. I will adjust the learning data set.

Thanks and Best regards.
